doc.go

```diff
@@ -3,14 +3,42 @@ package pty
 
 import (
 	"errors"
-	"os"
+	"io"
 )
 
 // ErrUnsupported is returned if a function is not
 // available on the current platform.
 var ErrUnsupported = errors.New("unsupported")
 
 // Opens a pty and its corresponding tty.
-func Open() (pty, tty *os.File, err error) {
+func Open() (Pty, Tty, error) {
 	return open()
 }
+
+type (
+	FdHolder interface {
+		Fd() uintptr
+	}
+
+	// Pty for terminal control in current process
+	// for unix systems, the real type is *os.File
+	// for windows, the real type is a *WindowsPty for ConPTY handle
+	Pty interface {
+		// Fd intended to resize Tty of child process in current process
+		FdHolder
+
+		// string writer is used to identify Pty and Tty
+		io.StringWriter
```

Comment:
This breaks compatibility with go1.13 and earlier. I forgot when this was added, but it is fairly recent. It would be better to redefine the interface to make sure we still work with older go versions.

Reply:
io.StringWriter was introduced in go1.12

see https://golang.org/pkg/io/#StringWriter

Reply:
The RISCV reference compiler is on go1.6. I didn't explore to see if there are newer version, but might as well keep the retro-compatibility.

Reply:
If you mean the support for riscv64, it's officially supported by go since go1.14.

Reply:
How do you think about it now?

Reply:
Cool idea, LGTM.